packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -714,6 +719,25 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
       onActiveItemChanged(items[index], index, ev);
     }
   }
+
+  private _getItemKey(item: any, itemIndex: number): string | number {
+    const { getKey } = this.props;
+
+    let itemKey: string | number | undefined = undefined;
+    if (item) {
+      itemKey = item.key;
+    }
+
+    if (getKey) {
```

Comment:
shouldn't these be if () {} else if () {} else {}?

Reply:
well, at least the first 2.

I could see the 3rd being fine, where if getKey returns undefined, you'd want to fall back to itemIndex.

Reply:
@dzearing that's basically a copy of the current logic in `List.tsx`, we might check `getKey` first